src/ifcgeom/IfcGeomIteratorImplementation.h

```diff
@@ -590,6 +494,112 @@ namespace IfcGeom {
         const gp_XYZ& bounds_max() const { return bounds_max_; }
 
 	private:
+		void addRepresentationsFromContextIds() {
+			for (auto context_id : settings.context_ids()) {
+				IfcSchema::IfcGeometricRepresentationContext* context = ifc_file->instance_by_id(context_id)->as<IfcSchema::IfcGeometricRepresentationContext>();
```

Comment:
`as<>()` returns 0 pointer when it isn't the correct type. You might want do a `if (context)` afterwards to test for this and either upgrade to an exception (which can be caught in Python as it is propagated by swig) or use Logger::Error and just skip this id for ex.